File: PersonalWebhook.Server/IncomingRequestBase.cs
namespace PersonalWebhook.Server
{
    public class IncomingRequestBase(string sessionId)
    {
        public string Method { get; set; } = string.Empty;
        public string RequestId { get; set; } = Guid.NewGuid().ToString();
        public string SessionId { get; set; } = sessionId;
        public string RequestBody { get; set; } = string.Empty;
        public DateTime Date { get; set; } = DateTime.Now;
        public List<object> Headers { get; set; } = [];
        public string AbsoluteUri { get; set; } = string.Empty;
        public string QueryString { get; set; } = string.Empty;
        public string RemoteIpAddress { get; set; } = string.Empty;
        public int RemotePort { get; set; }
    }
}

File: personalwebhook.client/src/app/models/IncomingRequestBase.ts
export interface IncomingRequestBase {
  method: string;
  requestId: string;
  sessionId: string;
  requestBody: string;
  date: Date;
  headers: Array<DictionaryKeyValues>;
  absoluteUri: string;
  queryString: string;
  remoteIpAddress: string;
  remotePort: number;
}

export interface DictionaryKeyValues{
  key: string;
  value: string;
}
